### ISSUE TITLE
Port to nano/pico

### ISSUE THREAD

**Issue by Ebedthan** [bioSyntax/bioSyntax#13]
Hey everybody

I have choose to invest my last night in writing syntax highlighting for nano/pico. I an writing the manual for develop nano syntax highlighting. 

Take a look at what i have done so far. 
![image](https://user-images.githubusercontent.com/17315349/34567206-2b77e99e-f159-11e7-8892-b576e902def4.png)
I thinks i will finish porting tonight.

**Comment by ababaian** [comment]
Looks great @Ebedthan, can you start a git-branch called `nano` while you're working on this so that things are not pushed to master (which is for releases). Once all core-nano syntaxes are finished we can merge the nano branch with master.

**Comment by Ebedthan** [comment]
ok @ababaian

**Comment by Ebedthan** [comment]
ok @ababaian

**Comment by Ebedthan** [comment]
@ababaian i think we can't define new colors in nano. My research don't gave me results as expected. If you have any informations on how to add colors in nano for syntax highlighting, please let me know.
So far with colors set in nano we can't define colors for many files.

**Comment by ababaian** [comment]
It's probably ANSI escape colors and/or System colors. Less and even terminal-vim has the same issue but even partial coloring with 16 colors is much better then nothing at all.

We just have to work within the framework of what is possible. With just Red / Green / Blue / Orange / White / Black and Bold / Italics and Background / Foreground you can accomplish about 90% of the functionality of bioSyntax. In the colors excel spreadsheet look at the last columns for ANSI/System colors where they are applicable.

**Comment by Ebedthan** [comment]
Ok I take a look.

**Comment by Ebedthan** [comment]
Please @ababaian could you let me know the link to the excel spreadsheet ?

**Comment by ababaian** [comment]
`dev/theme/bioSyntax_theme.xls` [link](https://github.com/bioSyntax/bioSyntax/tree/master/dev/theme)

**Comment by ababaian** [comment]
How is it coming along @Ebedthan? Do you have guidelines / instructions on the work-flow?

**Comment by Ebedthan** [comment]
Hey @ababaian 

Yes i have instructions. 
First you need to create a `*.nanorc` file defining syntax highlighting.  
Secondly you need to copy this file to `/usr/share/nano/` and to `~/.nano/syntax/` (the first path should also contains  `.nanorc` files by default for others languages such as php, perl, etc., they don't need to be copied to `~/.nanorc/` file but the following step should be done to see colors).  
Thirdly you need to add path to these files by creating/editing `~/.nanorc` doing `nano ~/.nanorc` and add `include /usr/share/nano/*.nanorc`.  
After that you will be able to see colors in your nano/pico editor.

**Comment by ababaian** [comment]
KK Anicet, I've copied your instructions to the top of the post to organize the information a bit. Keep us all updated about your progress with the pico syntaxes or if you run into trouble :+1:

**Comment by Ebedthan** [comment]
ok @ababaian

**Comment by Ebedthan** [comment]
Hello everybody, 
I have create the `nano` branch. Add fasta and fastq.nanorc files. Follow instructions above in the first comment to test it on your computer.

**Comment by ababaian** [comment]
Hows it coming along Anicet? Maybe one of the other @bioSyntax/devs can help out? This syntax format looks pretty tricky with the limitations. Have you found any good online guides on writing in-depth syntax files?

**Comment by Ebedthan** [comment]
Yes @ababaian i have uploaded works i'vé already finish on the nano branch. Take a look. I will upload guidelines today.

**Comment by Ebedthan** [comment]
![image](https://user-images.githubusercontent.com/17315349/35035002-e040293c-fb67-11e7-96be-a944eb64f323.png)
Hey everybody


![image](https://user-images.githubusercontent.com/17315349/35035040-f70d48e8-fb67-11e7-9eb7-3760ac392833.png)

![image](https://user-images.githubusercontent.com/17315349/35035144-509afca2-fb68-11e7-82c7-069b4ecaffb7.png)

I have finished with bed, fasta and fastq syntax. If anyone wants to help for the others files types it would be very nice. 
The limitations of colours is really a problem.

**Comment by Ebedthan** [comment]
Some helpful links 
https://gist.github.com/mattcurrycom/9d692d5dbb758e0724881e2c4fb03ef4
https://github.com/AbhishekGhosh/nano-syntax-highlighting-iNano-/
https://askubuntu.com/questions/90013/how-do-i-enable-syntax-highlighting-in-nano

**Comment by Ebedthan** [comment]
Hey @ababaian, since Valid colors in nano are white, black, red, blue, green, yellow, magenta, cyan.
 And for foreground colors, the use of the prefix "bright" to get a stronger highlight are the only ways to define syntax highlighting, i have found difficult to define syntax for sam, gtf, pdb files for examples.

**Comment by ababaian** [comment]
Start with GTF which is the simplest from that list. Can you get a Robust Column Selection regular expressions defined, that's more important then all the color details.

If you're able to write the logic in nano to arbitrarily select columns I'll be able to define a color scheme which works within the palette available. It may not be a perfect match but `less` only uses about 8 colors too and it works. When you can exploit 8 colors + bright + bg/fg selections you end up with lots of choices.

**Comment by alyeffy** [comment]
#2 task 8
